tests/herbie.egg

```diff
@@ -0,0 +1,204 @@
+;; Implements the simplification layer of herbie in egg-smol🫡
+
+;; Bools!
+(datatype B (True) (False))
+(function or (B B) B)
+(rewrite (or (True) x) (True))
+(rewrite (or x (True)) (True))
+(rewrite (or (False) (False)) (False))
```

Comment:
Could we replace these with primitive bools (once we add them)? Or are you using their eq-ability?